src/private/WRLDRoutingServiceHelpers.mm

```diff
@@ -0,0 +1,80 @@
+#include "WRLDRoutingServiceHelpers.h"
+
+#import "WRLDRoute.h"
+#import "WRLDRouteSection.h"
+#import "WRLDRouteDirections.h"
+#import "WRLDRouteStep.h"
+#import "WRLDRouteTransportationMode.h"
+
+#import <CoreLocation/CoreLocation.h>
+
+namespace Eegeo
+{
+    namespace Routes
+    {
+        namespace Webservice
+        {
+            namespace Helpers
+            {
+                WRLDRoutingQueryResponse* CreateWRLDRoutingQueryResponse(const Eegeo::Routes::Webservice::RoutingQueryResponse& result)
+                {
+                    WRLDRoutingQueryResponse* routingQueryResponse = [[WRLDRoutingQueryResponse alloc] init];
+                    [routingQueryResponse setSucceeded:result.Succeeded];
+                    routingQueryResponse.results = [[NSMutableArray alloc] initWithCapacity:(result.Results.size())];
+                    for(auto& route : result.Results)
+                    {
+                        WRLDRoute* wrldRoute = [[WRLDRoute alloc] init];
+                        wrldRoute.sections = [[NSMutableArray alloc] initWithCapacity:(route.Sections.size())];
+                        for(auto& section : route.Sections)
+                        {
+                            WRLDRouteSection* routeSection = [[WRLDRouteSection alloc] init];
+                            routeSection.steps = [[NSMutableArray alloc] initWithCapacity:(section.Steps.size())];
+                            for(auto& step : section.Steps)
+                            {
+                                int pathCount = static_cast<int>(step.Path.size());
+                                CLLocationCoordinate2D* path = new CLLocationCoordinate2D[pathCount];
+
+                                for (int i=0; i<pathCount; i++)
+                                {
+                                    path[i] = CLLocationCoordinate2DMake(step.Path[i].GetLatitudeInDegrees(), step.Path[i].GetLongitudeInDegrees());
+                                }
+
+                                const Eegeo::Routes::Webservice::RouteDirections& directions = step.Directions;
+                                WRLDRouteDirections* routeDirections = [[WRLDRouteDirections alloc] init];
+                                [routeDirections setType: [NSString stringWithCString: directions.Type.c_str() encoding:NSUTF8StringEncoding]];
```

Comment:
Talked this over in the formal review - understand this is what the poiService API is doing as well, but believe we want to look at initializing these via [initWith] methods rather than setters, since the result type is expected to be immutable.